### PR TITLE
[MRG] update gather implementations for LCA and SBT to use thresholding

### PR DESCRIPTION
**Pull request by ctb**
This is for optimization/performance improvement purposes; ref #838.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
My main comment after this experience is that we need some tests to check that threshold_bp is being passed in properly to both LCA and SBT gather... or maybe generically for Index objects...

File: sourmash/lca/lca_utils.py [hunk 1]
```diff
@@ -317,12 +317,19 @@ def search(self, query, *args, **kwargs):
...
 
     def gather(self, query, *args, **kwargs):
+        if not query.minhash:
+            return []
+
         results = []
-        for x in self.find_signatures(query.minhash, 0.0,
+        threshold_bp = kwargs.get('threshold_bp', 0.0)
+        threshold = threshold_bp / (len(query.minhash) * self.scaled)
+        print('xxx', threshold)
```
Comment (luizirber) [inline sourmash/lca/lca_utils.py]: This `print` is really messing the output
```suggestion
```

File: sourmash/lca/lca_utils.py [hunk 2]
```diff
@@ -411,26 +418,18 @@ def find_signatures(self, minhash, threshold, containment=False,
...
-            debug('count: {}; query_mins: {}; match size: {}',
-                  count, len(query_mins), match_size)
-
             if containment:
                 score = count / len(query_mins)
             else:
                 score = count / (len(query_mins) + match_size - count)
 
-            debug('score: {} (containment? {})', score, containment)
-
```
Comment (luizirber) [inline sourmash/lca/lca_utils.py]: Aren't these `debug()` statements useful?

File: sourmash/sbt.py [hunk 3]
```diff
@@ -312,6 +312,13 @@ def gather(self, query, *args, **kwargs):
         # use a tree search function that keeps track of its best match.
         search_fn = GatherMinHashes().search
 
+        leaf = next(iter(self.leaves()))
+        tree_mh = leaf.data.minhash
+        scaled = tree_mh.scaled
```
Comment (luizirber) [inline sourmash/sbt.py]: Oh, the classic "what queries does this tree support" trick =P

**Comment by luizirber** [comment]
Big improvements for LCA search and gather, not much difference for SBTs... (This PR is the `gather` on the X-axis)

LCA:
image
image

SBT:
image
image

File: sourmash/sbt.py (hunk 3 above)
Comment (ctb) [reply sourmash/sbt.py]: <sigh>

File: sourmash/lca/lca_utils.py (hunk 1 above)
Comment (ctb) [reply sourmash/lca/lca_utils.py]: fixed separately

File: sourmash/lca/lca_utils.py (hunk 2 above)
Comment (ctb) [reply sourmash/lca/lca_utils.py]: Fixed.

**Comment by ctb** [comment]
I think I've addressed all the review issues?

**Pull request merged**